src/classes/ClassWithStudentId.cls

```diff
@@ -0,0 +1,22 @@
+public with sharing class ClassWithStudentId{
+    /**
+    method that returns a map where key is class and value is set of studentid in it. 
+    */
+    public static Map<Class__c,Set<Student__c>> getClassesWithStudentId(){
+        List<Class__c> myClasses = [SELECT id,(SELECt id FROM Students__r) FROM Class__c]; 
+        Map<Class__c,Set<Student__c>> classMap = new Map<Class__c,Set<Student__c>>();
+        for(Class__c myClass : myClasses){
+            List<Student__c> studentList = myClass.Students__r;
+            Set<Student__c> studentsSet=new Set<Student__c>();
+            //iterate studentList and put each student record in a set
+            for(Student__c student : studentList){
```

Comment:
Rather than iterating the for loop, you can use the below statement:
`Set<Student__c> studentsSet = new Set<Student__c>(myClass.Students__r);`. For more details follow the apex doc:
https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_methods_system_set.htm